lib/imports/api/authority-registry/server/authority-registry.js

```diff
@@ -0,0 +1,319 @@
+      }
+    }
+    // remove servers that have failed too many health checks
+    if (failedConnectionsAboveThreshold.length > 0) {
+      try {
+        removeAuthorities.call({ _ids: failedConnectionsAboveThreshold });
+        console.log(`ProseMeteor AuthorityRegistry: Removed ${failedConnectionsAboveThreshold.length} authorities from the registry because of failed health checks`);
+      } catch (e) {
+        console.log(`ProseMeteor AuthorityRegistry: Failed to remove authorities ${failedConnectionsAboveThreshold}: ${e}`);
+      }
+    }
+
+    // return total number of bad connections, combining failedConnections with failedConnectionsAboveThreshold
+    return failedConnections.concat(failedConnectionsAboveThreshold).length;
+  }
+  /**
+  *  Returns the full authority registry from the database.
+  * @returns {Array} full registry
+  */
+  fullRegistry () {
+    return authorityRegistryColl.find({}).fetch();
+  }
+
+  /**
+  *  Returns a single authority from the registry database, found via its url.
+  * @param {Object} params
+  * @param {String} params.authorityUrl     full url of the authority to find (protocol:ip:port)
+  * @returns {{}} authority
+  */
+  getAuthorityInfoByUrl ({ authorityUrl }) {
+    return authorityRegistryColl.findOne({ fullUrl: authorityUrl });
+  }
+
+  /**
+  *  Returns a single authority from the registry database, found via a docId that lives in an authority.
+  * @param {Object} params
+  * @param {String} params.docId        id of the doc that the wanted authority is hosting
+  * @returns {{}} authority
+  */
+  getAuthorityInfoByDocId ({ docId }) {
+    return authorityRegistryColl.findOne({ docIds: docId });
+  }
+
+  /**
+  *  Returns a single authority from the registry database, found via a groupId that lives in an authority.
+  * @param {Object} params
+  * @param {String} params.groupId        groupId that the wanted authority is hosting
+  * @returns {{}} authority
+  */
+  getAuthorityInfoByGroupId ({ groupId }) {
+    return authorityRegistryColl.findOne({ groupIds: groupId });
+  }
+  /**
+  *  Returns if the specified server is registered or not.
+  * @param {Object} params
+  * @param {String} params.ip     ip of the server to remove from the registry
+  * @param {Number} params.port   port of the server to remove from the registry
+  * @returns {Boolean} if the server is registered
+  */
+  isRegistered ({ ip, port }) {
+    check(ip, String);
+    check(port, Number);
+    return authorityRegistryColl.find({ ip, port }).count() !== 0;
+  }
+
+  /**
+  *  Adds a doc to an authority in the registry.
+  * @param {Object} params
+  * @param {String} params.docId      unique id for this doc
+  * @param {String} [params.groupId]  optional, a groupId of this doc to associate it with other docs
+  */
+  addDocToAuthority ({ docId, groupId = null }) {
+    addDocToAuthority.call({ docId, groupId, fullUrl: this.fullUrl });
```

Comment:
I agree with this. Unless there is very good reason for this it could lead to scaling problems.